### PR TITLE
Add design evaluation based on design principles to TEP template

### PR DESCRIPTION
**Pull request by jerop**
Add a section in the TEP template where the authors can discuss how their proposals affects the reusability, simplicity, flexibility and conformance of Tekton, as described in the recently-added design principles

https://github.com/tektoncd/community/blob/master/design-principles.md

/cc @vdemeester @bobcatfish @afrittoli @ImJasonH

### CODE REVIEW

**Review by vdemeester — APPROVED** [review]
/meow

File: teps/NNNN-tep-template/README.md [hunk 1]
```diff
@@ -238,6 +238,12 @@ All code is expected to have adequate tests (eventually with coverage
 expectations).
 -->
 
+# Design Evaluation
+<!--
+How does this proposal affect the reusability, simplicity, flexibility 
+and conformance of Tekton, as described in [design principles](../../design-principles.md)
```
Comment (bobcatfish) [inline teps/NNNN-tep-template/README.md:244]: total nitpick but you might want to put the full link e.g. https://github.com/tektoncd/community/blob/master/design-principles.md cuz i dont think the markdown link in the comment will get rendered

**Review by bobcatfish — COMMENTED** [review]
Thanks @jerop!

/lgtm

So @afrittoli @ImJasonH @abayer can take a look:
/hold

File: teps/NNNN-tep-template/README.md (hunk 1 above)
Comment (jerop) [reply teps/NNNN-tep-template/README.md]: updated the link

**Review by vdemeester — APPROVED** [review]

**Comment by jerop** [comment]
@afrittoli @ImJasonH @abayer please take a look so that new TEPs evaluate how the proposals affect the reusability, simplicity, conformance and flexibility of Tekton

**Comment by ImJasonH** [comment]
/lgtm

🎉

**Review by afrittoli — COMMENTED** [review]
Thank you!
/lgtm

**Pull request merged**